src/library_dylink.js

```diff
@@ -181,7 +181,7 @@ var LibraryDylink = {
         } else if (typeof value === 'number') {
           GOT[symName].value = value;
         } else {
-          assert(false, 'bad export type for `' + symName + '`: ' + (typeof value));
+          throw new Error('bad export type for `' + symName + '`: ' + (typeof value));
```

Comment:
How about `abort(..)` to replace `assert(false, ..)`?

Reply:
I guess it depends how likely it is that you would want to continue execution after attempting to dlopen a side modules that is ill-formed or in-compatible.   I feel like being able to continue is probably nicer.   abort generally means that whole program is over.

Reply:
Are you talking about just this one place or the other locations too?

Reply:
I meant in general - it seems like that is what `abort()` is meant for. Unless you want the ability for an assertion to fail but the program to keep running somehow? I think that sounds too dangerous to try to support, and not worthwhile.

Reply:
In this case the error would be in dlopen and it would be recoverable, yes.  

See https://github.com/emscripten-core/emscripten/blob/8d47073959cc9122b3a7dabcc8b60fea3165096e/src/library_dylink.js#L822-L830

Reply:
Ah, I see, thanks, so this is a special location then.

Yeah, throwing an error sounds right then.